endpointregistry/epregistry.go

```diff
@@ -0,0 +1,62 @@
+package endpointregistry
+
+import (
+	"sync"
+	"time"
+)
+
+type entry struct {
+	detected         time.Time
+	inflightRequests int64
+}
+
+var data map[string]entry = map[string]entry{}
+var m sync.Mutex
```

Comment:
Lets define a type and have methods on it. This way we can throw it away or have several of them (e.g. for tests, or to run multiple proxies and dataclients). In general try to avoid global variables.